SwiftWisdom/Core/Foundation/Time/Date+Extensions.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Date+Extensions.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 1/28/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+
+extension Date {
+    
+    static var ip_millisecondFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "Y-MM-dd HH:mm:ss.SSSS"
+        return formatter
+    }
+
+    /// Returns a string in the long date format, ex: January 14, 2015
+    var ip_longDateString: String {
+        let formatter = DateFormatter()
+        formatter.locale = Locale.current
+        formatter.dateStyle = .long // Ex: January 14, 2015
+        return formatter.string(from: self)
+    }
+
+    /// Returns a string in the short date format, ex: 1/14/2015
+    var ip_shortDateString: String {
+        let formatter = DateFormatter()
+        formatter.locale = Locale.current
+        formatter.dateFormat = "M.d.yyyy"
```

Comment:
I actually think that these should shift to using statically created formatters. Like `ip_millisecondTimeStampString` does. Because I'm pretty sure formatters are pretty intensive to create. This SO answer agrees: http://stackoverflow.com/questions/28504589/whats-the-best-practice-for-nsdateformatter-in-swift